java/analyze-hotspots/src/main/java/com/esri/arcgisruntime/sample/analyzehotspots/DateRangeDialogFragment.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2019 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.analyzehotspots;
+
+import android.app.AlertDialog;
+import android.app.DatePickerDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.DialogFragment;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.Button;
+import android.widget.DatePicker;
+import android.widget.EditText;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Locale;
+
+public class DateRangeDialogFragment extends DialogFragment {
+
+  private static final String TAG = DateRangeDialogFragment.class.getSimpleName();
+
+  private static final String ARG_TITLE = DateRangeDialogFragment.class.getSimpleName() + "_title";
+
+  private static final String ARG_BUTTON_TEXT = DateRangeDialogFragment.class.getSimpleName() + "_button_text";
+
+  private SimpleDateFormat mSimpleDateFormatter;
+
+  private Date mMinDate;
+
+  private Date mMaxDate;
+
+  private OnAnalyzeButtonClickListener mOnAnalyzeButtonClickListener;
+
+  private String title;
+
+  private String buttonText;
```

Comment:
Could we fix the inconsistent naming of fields as part of these changes as well? I think for samples we use `mXYZ` style. @TADraeseke please confirm.